crmd/utils.c

```diff
@@ -1069,9 +1069,40 @@ update_attrd_helper(const char *host, const char *name, const char *value, const
     }
 }
 
+#if !HAVE_ATOMIC_ATTRD
+static int
+update_without_attrd(const char *host_uuid, const char *name, const char *value, const char *user_name)
+{
+    if (fsa_cib_conn == NULL) {
+        return -1;
+    }
+
+    crm_trace("updating status for host_uuid %s, %s=%s", host_uuid, name ? name : "<null>", value ? value : "<null>");
+    if (value) {
+        return update_attr_delegate(fsa_cib_conn, cib_none, XML_CIB_TAG_STATUS, host_uuid, NULL, NULL,
+                                    NULL, name, value, FALSE, user_name, NULL);
+    } else {
+        return delete_attr_delegate(fsa_cib_conn, cib_none, XML_CIB_TAG_STATUS, host_uuid, NULL, NULL,
+                                    NULL, name, NULL, FALSE, user_name);
+    }
+}
+#endif
+
 void
 update_attrd(const char *host, const char *name, const char *value, const char *user_name, gboolean is_remote_node)
 {
+#if !HAVE_ATOMIC_ATTRD
+    /* Talk directly to cib for remote nodes if it's legacy attrd */
+    if (is_remote_node) {
+        /* host is required for updating a remote node */
+        CRM_CHECK(host != NULL, return;);
+        /* remote node uname and uuid are equal */
+        if (update_without_attrd(host, name, value, user_name) < pcmk_ok) {
+            crm_err("Could not update attribute %s for remote-node %s", name, host);
+        }
+        return;
+    }
+#endif
```

Comment:
This belongs in update_attrd_helper().

Wouldn't you need to handle command='C' somehow as well?

I"m surprised someone wants to use pacemaker_remote with heartbeat. That's not exactly a well tested use case ...